In [1]:
from scipy.optimize import minimize

# 1. Inputs

In [6]:
#our budget for the purchase
budget = 10000

#size of a container in m3
total_size = 50

#price of goods
g1_price = 300
g2_price = 600

#size of goods
g1_size = 2
g2_size = 1

#profit of each goods
g1_profit = 1200
g2_profit = 1000

# 2. Objective and constraints

In [7]:
def objective(x):
    g1, g2 = x
    return g1*g1_profit + g2*g2_profit

def constraint1(x):
    g1, g2 = x
    return -(g1*g1_price + g2*g2_price) + budget

def constraint2(x):
    g1, g2 = x
    return -(g1*g1_size + g2*g2_size) + total_size

# 3. Initial guess

In [8]:
x0 = [1,1]
print(-objective(x0))

-2200


In [10]:
cons1 = {'type': 'ineq', 'fun': constraint1}
cons2 = {'type': 'ineq', 'fun': constraint2}
cons = (cons1, cons2)

# 4. Solution

In [11]:
sol = minimize(lambda x: -1.0*objective(x),x0, method='SLSQP', constraints=cons)

print(sol)

     fun: -32222.222047310468
     jac: array([-1200., -1000.])
 message: 'Optimization terminated successfully'
    nfev: 6
     nit: 6
    njev: 2
  status: 0
 success: True
       x: array([22.22222214,  5.55555548])


Conclusion: 22.2 good n°1 and 5.5 good n°2 will give a gain of 32,222.22.